### PR TITLE
[terra-tabs] Fixed keyboard navigation for draggable tabs

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Fixed keyboard navigation for draggable tabs failing due to draggable props from react beautiful dnd.

**What was changed:**
Tab was focused programmatically to fix the issue.

**Why it was changed:**
Arrow Key navigation was failing because of the focus not being on the tab after mouse click.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9761 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-tabs/src/common-tabs/_Tab.jsx [hunk 1]
```diff
@@ -230,6 +230,10 @@ const Tab = ({
         onChange(event, itemKey);
       }
     }
+    // Fix for keyboard navigation after mouse click which was failing due to draggable props.
+    if (isDraggable) {
+      event.currentTarget.focus();
```
Comment (ShettyAkarsh) [inline packages/terra-tabs/src/common-tabs/_Tab.jsx:235]: so now the focus always remains on the dragged tab ? and once dropped the keyboard navigation starts from that position ?

**Comment by ShettyAkarsh** [comment]
@saket2403  does the arrow navigation works fine when the isdraggable prop is false ?

File: packages/terra-tabs/src/common-tabs/_Tab.jsx (hunk 1 above)
Comment (saket2403) [reply packages/terra-tabs/src/common-tabs/_Tab.jsx:235]: Before the change when a draggable tab was clicked, the focus was going onto the main or the body div due to the draggable props of the dnd library. So I have put the focus on the clicked tab. Change has nothing to do with whether the tab has been dragged or not.

**Comment by saket2403** [comment]
> @saket2403 does the arrow navigation works fine when the isdraggable prop is false ?

Yes it works fine in non draggable tabs because when tab is clicked the focus is on the tab.

**Review by supreethmr — APPROVED** [review]

**Pull request merged**